src/libhirte/common/time-util.c

```diff
@@ -0,0 +1,26 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+#include <time.h>
+
+#include "time-util.h"
+
+int64_t get_time_micros() {
+        struct timespec now;
+        if (clock_gettime(CLOCK_REALTIME, &now) < 0) {
+                return 0;
+        }
+        // NOLINTBEGIN(bugprone-narrowing-conversions, cppcoreguidelines-narrowing-conversions)
+        int64_t now_micros = now.tv_sec * sec_to_microsec_multiplier +
+                        now.tv_nsec * nanosec_to_microsec_multiplier;
+        // NOLINTEND(bugprone-narrowing-conversions, cppcoreguidelines-narrowing-conversions)
+        return now_micros;
+}
+
+int64_t finalize_time_interval_micros(int64_t start_time_micros) {
+        return get_time_micros() - start_time_micros;
+}
+
+double micros_to_millis_double(uint64_t time_micros) {
```

Comment:
I'd remove the `_double` suffix - its already indicated by the return type of the function.